src/ProjectTemplates/Shared/Project.cs

```diff
@@ -362,6 +355,51 @@ public async Task VerifyHasProperty(string propertyName, string expectedValue)
         Assert.Contains($"<{propertyName}>{expectedValue}</{propertyName}>", projectFileContents);
     }
 
+    public void SetCurrentRuntimeIdentifier()
```

Comment:
As an aside, seems like a gap that an API like this isn't in the runtime already.

Reply:
There is an API like this in the runtime: https://learn.microsoft.com/dotnet/api/system.runtime.interopservices.runtimeinformation.runtimeidentifier

However, that will get you the exact RID of the machine (ex. `win10-x64`), which is not what we want/need here. We need the portable RID (`win-x64`) here because that is what is emitted in the publish directory structure.

Reply:
Right I've run into this issue before when doing automation like this. My point was it would be good to have an API that gets the portable RID.